src/task/models/tileMergeTaskManager.ts

```diff
@@ -311,27 +365,79 @@ export class TileMergeTaskManager {
     part: UnifiedPart,
     zoom: number,
     tilesMetadata: MergeTilesMetadata,
-    parentSpan?: Span | undefined
+    parentSpan: Span | undefined,
+    skipTasksUntilIndex: number = 0
   ): AsyncGenerator<MergeTaskParameters, void, void> {
     const span = createChildSpan(`${TileMergeTaskManager.name}.${this.createTasksForPart.name}.zoom.${zoom}`, parentSpan);
-    span.setAttributes({ part: JSON.stringify(part), zoom, maxTileBatchSize: this.tileBatchSize });
+    span.setAttributes({
+      part: JSON.stringify(part),
+      zoom,
+      maxTileBatchSize: this.tileBatchSize,
+      skipTasksUntilIndex,
+    });
+
     const { layerRelativePath, grid, isNewTarget, tileOutputFormat } = tilesMetadata;
-    const logger = this.logger.child({ zoomLevel: zoom, isNewTarget, layerRelativePath, tileOutputFormat, grid });
+    const logger = this.logger.child({
+      zoomLevel: zoom,
+      isNewTarget,
+      layerRelativePath,
+      tileOutputFormat,
+      grid,
+      skipTasksUntilIndex,
+    });
 
     const footprint = part.footprint;
     const rangeGenerator = this.tileRanger.encodeFootprint(footprint, zoom);
     const batches = tileBatchGenerator(this.tileBatchSize, rangeGenerator);
     const sources = this.createPartSources(part, grid, layerRelativePath);
 
+    let localTaskCounter = 0; // Local counter for this zoom level
+
     for await (const batch of batches) {
-      logger.debug({ msg: 'Yielding batch task', batchSize: batch.length });
-      span.addEvent('Yielding batch task', { batchSize: batch.length, batch: JSON.stringify(batch) });
+      // Skip tasks if we're resuming and haven't reached the resume point yet
+      if (localTaskCounter < skipTasksUntilIndex) {
```

Comment:
feels like this `localTaskCounter` is unnecessary, its being initialized each time to 0 (line 396)
so condition will always be true

Reply:
why not just use the `skipTasksUntilIndex`, or even change to more meaningful var name and set it for the yeild action